tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -138,8 +138,10 @@ def _s3_has_all_data(self, num_messages):
         return False
 
     @cluster(num_nodes=4)
-    @matrix(num_messages=[2])
-    def test_restore_with_config_batches(self, num_messages):
+    @matrix(num_messages=[2],
+            cloud_storage_type=[CloudStorageType.ABS, CloudStorageType.S3])
```

Comment:
Does this style of parametrization definitely work?  I think the assumption is that test_context is getting updated, but looking at Parametrize.apply I don't see where that happens.

(if it works it works, I'm just saying it's new to me)

Reply:
I think it does. The tests ran fine [here](https://ci-artifacts.dev.vectorized.cloud/redpanda/0185bbd3-3dac-4379-94b1-a764f6a4a29d/vbuild/ducktape/results/final/report.html). Have a look at [Matrix.apply](https://github.com/confluentinc/ducktape/blob/fb6a86acc2e11653f464a2e5111c860a436256ea/ducktape/mark/_mark.py#L127).